src/components/buildQuizPage.ts

```diff
@@ -1,56 +1,133 @@
-    answerType.appendChild(answerOption);
-    let label_answerOption : HTMLElement = document.createElement('label');
-    label_answerOption.setAttribute('for','a');
-    label_answerOption.textContent = 'blub'
-    answerType.appendChild(label_answerOption);
-    questionContainer.appendChild(answerType);
-
-    /* submit button */
-    let submitButton : HTMLElement = document.createElement('button');
-    submitButton.className = 'btn submit hidden';
-    submitButton.setAttribute('type', 'button');
-    submitButton.textContent = 'SUBMIT';
-    questionContainer.appendChild(submitButton);
-    body.appendChild(questionContainer);
+    fill_template(questionCards[quiz.round])
+
+    /* using a new template after each submit */
+    let submitButton = document.querySelector('button');
+    submitButton?.addEventListener("click", function () {
+        changeButton();
+        quizMaster.handleQuizScore(quiz, collectSelectedAnswers());
+        
+        if (quiz.hasReachedEnd()) {
+            fill_result(quiz.score);
+        }
+        else {
+            fill_template(questionCards[quiz.round]);
+        }
+    });
+};
+
+/**
+ * after the quiz the result template will show up
+ */
+function fill_result(score: number) {
+    fetch('quizResult.hbs')
+        .then(response => {
+            if (!response.ok) {
+                throw new Error("no templates found")
+            }
+            return response.text();
+        })
+        .then(response => {
+            const template = compile(response);
+            const filled = template(score);
+            document.getElementById('quizData').innerHTML = filled;
+        });
+}
+
+/* fetch templates to quiz page */
+function fill_template(question: Question) {
+    let data = {
+        question: question.question,
+        answers: question.answers,
+    };
+
+    let questionTemplate: string = "null";
+    if (question.type === QuestionType.FREE_TEXT) {
+        questionTemplate = 'freeTextQuestion.hbs';
+    } else if (question.type === QuestionType.SINGLE_CHOICE) {
+        questionTemplate = 'singleChoiceQuestion.hbs';
+    } else if (question.type === QuestionType.MULTIPLE_CHOICE) {
+        questionTemplate = 'multipleChoiceQuestion.hbs';
+    }
+
+    fetch(questionTemplate)
+        .then(response => {
+            if (!response.ok) {
+                throw new Error("no templates found")
+            }
+            return response.text();
+        })
+        .then(response => {
+            const template = compile(response);
+            const filled = template(data);
+            document.getElementById('quizData').innerHTML = filled;
+        });
 };
 
-buildQuizPage();
\ No newline at end of file
+
+/* This method collect the answers for QuizMaster */
+export function collectSelectedAnswers(): string[] {
+    let collectedAnswers: string[] = [];
+    let currentAnswerType = document.querySelector('input').type;
+
+    if (currentAnswerType === 'text') {
```

Comment:
Each question knows its type. If you pass the question to collectSelectedAnswers() then it will not have to deduce which question type it is looking forward